src/t2iapi/device/types.proto

```diff
@@ -67,4 +67,13 @@ Also see the XML 1.0 specification.
 message ExpandedName {
     google.protobuf.StringValue uri = 1;
     string local_name = 2;
-}
\ No newline at end of file
+}
+
+/*
+Represents electrical power sources of the device.
+ */
+enum ElectricalPowerSource{
+    ELECTRICAL_POWER_SOURCE_INTERNAL = 0;   // INTERNAL ELECTRICAL POWER SOURCE as defined in IEC 60601-1:2005
+    ELECTRICAL_POWER_SOURCE_EXTERNAL = 1;   // an external reserve electrical power source providing electrical
```

Comment:
why "external reserve electrical"? maybe write "mains power"
(or maybe "electric energy from an external circuit" as written in BICEPS)

Reply:
not relevant anymore, changed the enum